### ISSUE TITLE
Useless interface, please make it better

### ISSUE THREAD

**Issue by Burgsch** [gorhill/uBlock#1310]
Hey, 

generally uBlock is better in terms of performance than Adblock but it is really crap when we mean interface.
The way of creating rules is not good, the same goes to "picker" - it is ages behind Adblock's "element hiding" feature.
There is lacking "open blocked items" on webpage. How can I find what blocked script makes a web page half-loading? This in-built system is just one pile of miserable krap.
It would be nice to have interface from Adblock or at least have same functionality.

**Comment by gorhill** [comment]
[CONTRIBUTING](https://github.com/gorhill/uBlock/blob/master/CONTRIBUTING.md).

**Comment by IDKwhattoputhere** [comment]
The only thing I ever missed from ABP is this window: ![](https://cloud.githubusercontent.com/assets/16030872/12659693/672c5ce8-c610-11e5-9586-90a69f212665.png)
It just has more options than uBO has in the corresponding window.

Everything else you describe is properly implemented in uBO.

**Comment by Burgsch** [comment]
It is not the same - in AB it is very easy to select object to hide. Element hiding is great tool. I can select frame, make selection wider, narrower depending of the hierarchy on the page.
In uBlock "picker" is too simple - you pick and advert thinks you clicked it - in AB clicks are no send to the browser but to AB panel.
uBlock's logger is too complicated compared to the "open blocked elements" in AB where I can easily choose what with what.
Generally speaking uBlock may have functionality, but the way of representing it and user-friendliness of the GUI is way below Adblock.

**Comment by gorhill** [comment]
> make selection wider

Just select another item in the suggested filter list to make the selection wider -- up to (but not including) the `body` element.

**Comment by IDKwhattoputhere** [comment]
The advert doesn't "think you clicked it". I don't know where you heard that.

**Comment by gorhill** [comment]
> In uBlock "picker" is too simple - you pick and advert thinks you clicked it 

@IDKwhattoputhere is right -- where does this come from? If you looked at the code you would see it's not possible, unless you have a repro case? There is an `iframe` covering the whole page when the element picker is in action, all clicks are captured by that frame.

**Comment by Burgsch** [comment]
> 
@IDKwhattoputhere is right -- where does this come from?
> 

I had an experience with a bit retarded ad - it was made with flash (swf). You selected picker, clicked on mentioned flash advert and instead of selecting it, there was new tab opened. Why? Because flash advert thought you clicked on it while you were using just picker.
There was no possibility to select frame on which flash ad was displayed.

In AB it never happens in element hiding mode which is way more convenient to use.
Don't get me wrong, uBlock is OK but in terms of GUI it is retarded compared to the AB, especially Element Hiding feature.

Finding which script of 30 blocked on the page is not big problem in AB. You have easy list what is being blocked, what is not. You click X icon and page reloads. Most of the time successfully at first tries.
How can I quickly find what script makes page broken here? Browsed Quick Manual but if that's it, it is really backward.

**Comment by gorhill** [comment]
It's not clear so far which blocker you are referring to with "AB" I will assume Adblock Plus/Firefox.

>  it was made with flash (swf).

This is a an edge case on FF: Flash-based applets float over the page, it's impossible for HTML to be rendered over these applets -- this includes uBO's `iframe`. At this point it's not worth trying to find a workaround for this -- Flash is on its way out, and it should always be in click-to-play mode anyways (in which case uBO would work fine). It works fine for Chromium though, the Flash applets are not floating over everything else.

> You have easy list what is being blocked, what is not

I disagree with ABP's approach of disabling filters: when someone does so, this disables a filter *everywhere*. If the blocking of a Google Analytics ("GA") script breaks a page, disabling the filter causes the blocking of GA to be disabled _everywhere_ -- not good. The proper solution is to override the block filter for the site only, while leaving the filter working everywhere else. This is what dynamic URL filtering allows, and it is a point-and-click feature of the [logger](https://github.com/gorhill/uBlock/wiki/The-logger#dynamic-url-filtering-rules).

Filtering expressions can be used in the logger so that it displays only the network requests which are blocked/allowed (i.e. `-- || ++`) -- to remove the noise.

**Comment by Burgsch** [comment]
We differ what a "convenient" interface meant, OK.
One thing is sure - I'm now trying to disable some frame via dynamic filter - picker.
I reload page, it comes back. It is just "###topMenu" - I selected it 4 times, it comes back.
In AB it happens only if webmaster made some kind of anti-feature like dynamic names ect. which require deeper edit of the basic filter.

You may defend your interface, or really have just another point of view (more dev eye) - but I have 3rd common user, few years AB experience eye.
And I clearly say AB is much easier to use, much better understanding what's going on, except filter edit which is not too good, but hey... you achieve something faster there than by uBlock.
You did good optimization and page is more responsive though.

**Comment by gwarser** [comment]
>One thing is sure - I'm now trying to disable some frame via dynamic filter - picker.
I reload page, it comes back. It is just "###topMenu" - I selected it 4 times, it comes back.
In AB it happens only if webmaster made some kind of anti-feature like dynamic names ect. which require deeper edit of the basic filter.

https://github.com/gorhill/uBlock/wiki/Element-picker#the-removed-element-reappears-when-you-reload-the-page

https://github.com/gorhill/uBlock/blob/master/CONTRIBUTING.md

**Comment by Burgsch** [comment]
Thanks, but after trying to unblock site's element I gave up.
It is waste of my time to do something that can be done few times faster, easier and more convenient in Adblock Plus, even for the price of performance.
Thread can be closed.

**Comment by IDKwhattoputhere** [comment]
@Burgsch @gwarser linked the CONTRIBUTING.md to inform you that you need to post additional information. The site where this is happening would be helpful for example.
If you don't wish to provide this information you can close the issue by yourself.
